doc/en/source/k8sIntegration.rst

```diff
@@ -0,0 +1,186 @@
+**********************
+Kubernetes Integration
+**********************
+
+The components of FogFlow can be built via source code as well as in docker environment using docker-compose tool. In docker environment each component of FogFlow is running on a single container. Whole FogFlow system will have to re-start in case any single component container goes down and if any single service is overloaded it cannot scale to handle the load.  
+To overcome these issues FogFlow would be migrated to Kubernetes. FogFlow components will be setup in Kubernetes cluster environment based on end user requirement. Kubernetes cluster can be deployed in various configurations:
+
+1.	Master and Worker on same Node
+2.	Single Master and Single Worker
+3.	Single Master and Multi Worker Node
+4.	Multiple Master and Multi Worker Node
+
+
+Along with cluster following features of K8s are implemented in FogFlow:
+1.	 High Availability and Load Balancing.
+2.	 Self-healing.
+3.	 Automated Rollouts & Rollback.
+4.	 Ease the deployment with Helm Support.
+
+Terminologies
+==============
+
```

Comment:
These are generic terms I think not required in user manual

Terminologies
==============

**High-Availability** and Load Balancing is about setting up Kubernetes, along with its supporting components in a way that there is no single point of failure. If the environment setup has multiple applications running on Single containers that container can easily fail. Same as the virtual machines for high availability in Kubernetes multiple replicas of containers can be run.
Load balancing is efficient in distributing incoming network traffic across a group of backend servers.
 A load balancer is a device that distributes network or application traffic across a cluster of servers. The load balancer has a big role to achieve high availability and performance increase of cluster.


**Self-healing** if any of the pod are deleted manually or a pod got deleted accidentally or restarted. The deployment will make sure that it brings back the pod because Kubernetes has a feature to auto-heal the pods.	


**Automated Rollouts & Rollback Rollouts** can be achieved by rolling update. Rolling updates are the default strategy to update the running version of your app. It updates cycles previous Pod out and bring newer Pod in incrementally.
When any introduced change that breaks production, then there should have a plan to roll back that change Kubernetes and kubectl offer a simple mechanism to roll back changes to resources such as Deployments.


**Helm** is a tool that streamlines installing and managing Kubernetes applications. It helps in managing Kubernetes applications. Helm Charts helps to define, install, and upgrade even the most complex Kubernetes application.
FogFlow document would be updated with the functioning details of above features to understand and access the Kubernetes environment well